[veehz] iP

veehz opened the pull request:
# Maid-Chan

> "Welcome Home" - Maid Chan (maybe) ([source](https://www.maidchan.com))

Modelled after a character in an [anime](https://sakurasou.fandom.com/wiki/Maid), Maid-Chan is

- Text-based
- ~Super~ _Smart_ and **Cool**

All you need to do is,

1. Download it from [here](https://github.com/veehz/ip/releases)
2. double-click it
3. add your tasks
4. chat with it :100: 

Future work:

- [X] Delete tasks
- [ ] Reminders (coming 

---

The `MaidChan` class is written like:

```java
public class MaidChan {
    private static TaskList taskList;
    private static Ui ui;
    private static Storage storage;
    private static Parser parser;
    // methods omitted
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

veehz committed:
.gitignore: Add rule to ignore class files

veehz committed:
docs: Add AI usage to AI.md

veehz committed:
refactor: Rename project from Duke to MaidChan and update related files

veehz committed:
Skeletal version that simply greets and exits

veehz committed:
AI: Observations on Copilot

veehz committed:
Change printing style and put message printing in sendMessage method

veehz committed:
Echos messages until user says bye

veehz committed:
Support for adding tasks and listing tasks

veehz committed:
Introducing Task class and updating task handling

veehz committed:
Allow mark and unmark tasks

veehz committed:
Add task types: ToDo, Deadline, and Event with parsing logic

veehz committed:
use scanner instead of console to read any kind of stdin

veehz committed:
gitignore: shell.nix for nix package manager

veehz committed:
Update expected output and input for test running

veehz committed:
Extract input handling to its own method

veehz committed:
Add exceptions

veehz committed:
Throw and handle TaskException and CommandNotFoundException

veehz committed:
Update testcases to test exception handling

veehz committed:
Checkstyle and JavaDoc

veehz committed:
Add delete task feature while grouping similar tasks together.

veehz committed:
Deal with invalid task numbers

veehz committed:
Refactor to use enums. Certain error messages are clearer.

veehz committed:
Task class: Add toRepr which mimics string for constructor

Let's create toRepr() method which mimics string that is accepted by constructor.

Let's create a constant command name as a static
field in the subclasses to prevent duplicate
hardcoded command names

veehz committed:
Implement load and save functionality

Let's implement load and save functionality which will
use the text file ./data/tasks.txt which saves it in
toRepr() form

veehz committed:
Merge branch 'branch-Level-7'

veehz committed:
Add TaskDateTime class for parsing and formatting date/time strings

veehz committed:
Task classes: Refactor to use TaskDateTime for date/time handling

Let's use TaskDateTime to deal with time instead of strings.

Let's change input and expected output for testing.

veehz committed:
Merge branch 'branch-Level-8'

veehz committed:
Refactor to extract Ui, Storage, Parser, TaskList

veehz committed:
Merge branch 'branch-A-MoreOOP'

veehz committed:
.gitignore: Exclude VSCode files

veehz committed:
Merge branch 'add-gradle-support'

veehz committed:
build.gradle: Change Duke to MaidChan

veehz committed:
Organize all files into package maidchan

veehz committed:
Add unit tests for Parser, Task, and Ui classes

Let's add unit tests for Parser, Task, and Ui classes.

Let's update build.gradle to support JUnit
parameterized tests.

Let's update AI usage in AI.md

veehz committed:
Add JavaDoc comments

veehz committed:
Add Checkstyle plugin

veehz committed:
Add Google Style Guide

veehz committed:
Refactor According to Style Guide

Let's adjust google_checks.xml to fit with Course Style Guide: https://se-education.org/guides/conventions/java/intermediate.html

Let's refactor code according to the style guide

veehz committed:
Add FindCommand to find matching tasks

veehz committed:
Merge branch 'branch-A-JavaDoc'

veehz committed:
Merge branch 'branch-A-CodingStandard'

veehz committed:
Merge branch 'branch-Level-9'

veehz committed:
Fix Style File and Refactor

Let's fix problems with styles that occurred due to AI hallucination.

Let's refactor based on style file.

veehz committed:
Add CS2103T-specific checkstyle files

veehz committed:
Refactor imports and add missing JavaDoc comments

veehz committed:
Merge branch 'branch-A-CheckStyle'

veehz committed:
Add GUI components for MaidChan

Currently uses CLI.

GUI is easier for a lot of users to use.

Let's create GUI components for MaidChan.

veehz committed:
Refactor GUI to use FXML

veehz committed:
Remove outdated text UI test files and scripts

veehz committed:
Refactor command execution methods for GUI

Commands uses Ui.sendMessage to send message
when using a CLI.

Let's remove Ui completely and refactor execution
code to use getResponse which works with the GUI.

veehz committed:
Add exit command handling to MaidChan

veehz committed:
Merge branch 'branch-Level-10'

veehz committed:
Add bounds checking for task index in command execution

Zonnie-23 commented on src/main/java/maidchan/task/TaskDateTime.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public class TaskDateTime {
+    public static final String ACCEPTED_FORMATS = "yyyy-MM-dd HH:mm or yyyy-MM-dd";
+    private boolean hasTime;
```
Nice use of boolean sounding variable

Zonnie-23 reviewed (commented):
Overall, code was well organised and easily understandable. OOP was well planned and executed. JavaDoc was also comprehensive. Error messages are also insightful

Zonnie-23 commented on src/main/java/maidchan/MaidChan.java:
```diff
@@ -0,0 +1,58 @@
+     * Constructs a new MaidChan bot.
+     */
+    public MaidChan() {
+        storage = new Storage(System.getProperty("user.dir") + File.separator + "data"
```
I am not sure, but should the System class be imported?

veehz replied:
Oh, this is interesting. I've never wondered that myself.

It doesn't need to be imported because according to the [Java Specification](https://docs.oracle.com/javase/specs/jls/se7/html/jls-7.html),

> A compilation unit automatically has access to all types declared in its package and also automatically imports all of the public types declared in the predefined package `java.lang`.

And [`System`](https://docs.oracle.com/javase/8/docs/api/java/lang/System.html) is defined under `java.lang`.

Zonnie-23 commented on src/main/java/maidchan/storage/Storage.java:
```diff
@@ -0,0 +1,99 @@
+     *
+     * @param tasks the list of tasks to save
+     */
+    public void saveTodoList(List<Task> tasks) {
```
nice use of verbs to name functions

Zonnie-23 reviewed (commented):
Just to add a few more comments

veehz committed:
Refactor task loading logic into a separate method for improved readability

veehz committed:
Add assertion for unknown task type during file loading

veehz committed:
Add assertion to ensure task description is not empty

veehz committed:
Add assertion to validate task index in removeTask method

veehz committed:
Merge pull request #1 from veehz/branch-A-Assertions

Add assertions

veehz committed:
Merge branch 'master' into branch-A-CodeQuality

veehz committed:
Merge branch 'branch-A-CodeQuality'

veehz committed:
Add Note and NoteList classes

There are only functionality for tasks.

Adding functionality for notes will allow users to
record miscellaneous information.

Let's create Note and NoteList classes to store a note
and a list of notes.

veehz committed:
Add Note-related commands

There is support for notes with Note and NoteList classes.

Let's
* add command for users to add notes
* add command for users to delete notes
* add command for users to list and view notes.

veehz committed:
Add note command handling to Parser

Note functionality is complete but not shown to user.

Let's expose the feature to users by adding it to Parser.

veehz committed:
Merge branch 'branch-D-Notes'

veehz committed:
Update user guide and add UI screenshot

veehz committed:
Fix method documentation in Task class
